### ISSUE TITLE
Validation and test on multiple datasets

### ISSUE THREAD

**Issue by chuong** [open-mmlab/mmdetection#3049]
**Describe the feature**
Currently training on multiple dataset is supported in this way

```
dataset_A_train = dict()
dataset_B_train = dict()
dataset_A_val = dict()
dataset_A_test = dict()

data = dict(
    imgs_per_gpu=2,
    workers_per_gpu=2,
    train = [
        dataset_A_train,
        dataset_B_train
    ],
    val = dataset_A_val,
    test = dataset_A_test
    )
```

When trying multiple validation or test datasets like this: 

```
dataset_A_train = dict()
dataset_B_train = dict()
dataset_A_val = dict()
dataset_A_test = dict()
dataset_B_test = dict()

data = dict(
    imgs_per_gpu=2,
    workers_per_gpu=2,
    train = [
        dataset_A_train,
        dataset_B_train
    ],
    val = dataset_A_val,
    test= [
        dataset_A_test,
        dataset_B_test
    ],
    )
```
I got an error:
```
Traceback (most recent call last):
  File "tools/test.py", line 153, in <module>
    main()
  File "tools/test.py", line 94, in main
    cfg.data.test.test_mode = True
AttributeError: 'list' object has no attribute 'test_mode'
```

**Motivation**
Support for multiple datasets is expected to work for validation and testing, just as for training.

**Comment by ZwwWayne** [comment]
Validation or test on multiple dataset is not supported currently, we will support that in the next release.

**Comment by ZwwWayne** [comment]
Fixed in #3522

**Comment by DimTrigkakis** [comment]
I think this is a case of less-than, since it just allows us to combine datasets into one and get average performance. If we have a small dataset to evaluate overfitting, a target dataset A and a target dataset B and we want to see how the training generalizes differently from A to B, while watching out for overfitting, we cannot evaluate all at the same time.
